pkgs/by-name/in/invoiceplane/package.nix

```diff
@@ -0,0 +1,84 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  nixosTests,
+  fetchYarnDeps,
+  fixup-yarn-lock,
+  nodejs,
+  yarn,
+  php,
+}:
+
+php.buildComposerProject (finalAttrs: {
+  pname = "invoiceplane";
+  version = "0-unstable-2024-10-30";
+
+  src = fetchFromGitHub {
+    owner = "InvoicePlane";
+    repo = "InvoicePlane";
+    #rev = "refs/tags/v${finalAttrs.version}";
+    rev = "5af4e5fa6198a194c2304a0349257a6f82e2c1fd";
+    hash = "sha256-P/chNhICFjAbOiwyL/BGhZ1dDLtPUEW0uD7OhAZEb8A=";
+  };
+
+  vendorHash = "sha256-k7YBs6x/ABNTHPx9/EZXa4W9kcLQqpruV7YFGAADZq0=";
+
+  nativeBuildInputs = [
+    fixup-yarn-lock
+    nodejs
+    yarn
+  ];
+
+  offlineCache = fetchYarnDeps {
+    yarnLock = "${finalAttrs.src}/yarn.lock";
+    hash = "sha256-8WIATiQmC4M7kMh3HyU3Av3t2SONp7hn278J/oHYABA=";
+  };
+
+  # Upstream composer.json file is missing the name, description and license fields
+  composerStrictValidation = false;
+
+  configurePhase = ''
+    runHook preConfigure
+
+    export HOME=$(mktemp -d)
+    yarn config --offline set yarn-offline-mirror $offlineCache
```

Comment:
You might be mixing the old yarn tools with the new ones. I would try to closely follow https://github.com/NixOS/nixpkgs/blob/9c85f9aa686888ac591d98d136eef71c5253fa72/doc/languages-frameworks/javascript.section.md#yarn-javascript-yarn

Also I would look at hedgedoc which does it super hacky though

Reply:
Thank you for the hint! Unfortunately still the same issue :(

Reply:
Maybe @lelgenio can help :)

Reply:
:thinking: 

I think their `yarn.lock` is out of sync with with their `package.json`.

`package.json` requires `^9.8.8`
https://github.com/InvoicePlane/InvoicePlane/blob/5af4e5fa6198a194c2304a0349257a6f82e2c1fd/package.json#L15

`yarn.lock` has an entry for `^9.8.6` (the key should match exactly what's in `package.json`)
https://github.com/InvoicePlane/InvoicePlane/blob/5af4e5fa6198a194c2304a0349257a6f82e2c1fd/yarn.lock#L212

Sure enough, if you clone the repo and run `yarn install`, `yarn.lock` will be updated with the correct version of `autoprefixer` and many other dependencies.

For now you'll need to store a local copy of this updated `yarn.lock`. But this could/should also be fixed upstream.

```nix
  offlineCache = fetchYarnDeps {
    yarnLock = ./yarn.lock;
    hash = "sha256-t9GYFqKY3YZ/l5LZXIHIgehgvxIbMMuYLwXvvdQaryM=";
  };

  preConfigure = ''
    cp ${./yarn.lock} ./yarn.lock
  '';
```

Reply:
Thank you @lelgenio this worked :+1: